lib/vrl/parser/src/parser.lalrpop

```diff
@@ -294,6 +301,7 @@ Compare: Expr = {
 };
 
 Add: Expr = {
+    Op<Add, "|", Factor>,
```

Comment:
Might want to move this into a separate rule named `Merge` just for consistency?